docs/book/how-to/customize-docker-builds/use-a-prebuilt-image.md

```diff
@@ -0,0 +1,122 @@
+
+This is the case where you are running a pipeline for the first time with ZenML and don't want ZenML to build an image for you. Here, you will have to provide an image to the `parent_image` attribute which already has all of your code and the right dependencies installed.
+
+{% hint style="info" %}
+Note that this is different from the case where you [only specify a parent image](../../../../docs/book/how-to/customize-docker-builds/docker-settings-on-a-pipeline.md#using-a-pre-built-parent-image) and don't want to `skip_build`. In the latter, ZenML still builds the image but does it on top of your parent image and not the base ZenML image.
+{% endhint %}
+
+### What should your image contain
+
+It is crucial that you ensure that all of your code files and dependencies are included in the image you provide to `DockerSettings` as parent image. This is because ZenML expects to be able to execute this image right away, without any modifications, to run your pipeline steps.
+
+The following points are derived from how ZenML builds an image internally and will help you make your own images:
+
+#### Your stack requirements
+
+A ZenML Stack can have different tools and each comes with its own requirements. You need to ensure that your image contains them. The following is how you can get a list of stack requirements.
+
+```python
+from zenml.client import Client
+
+stack_name = <YOUR_STACK>
+# set your stack as active if it isn't already
+Client().set_active_stack(stack_name)
+
+# get the requirements for the active stack
+active_stack = Client().active_stack
+stack_requirements = active_stack.requirements()
+```
+
+#### Integration requirements
+
+For all integrations that you use in your pipeline, you need to have their dependencies installed too. The following is how you can get a list of them.
+
+```python
+from zenml.integrations.registry import integration_registry
+from zenml.integrations.constants import HUGGINGFACE, PYTORCH
+
+# define a list of all required integrations
+required_integrations = [PYTORCH, HUGGINGFACE]
+
+# Generate requirements for all required integrations
+integration_requirements = set(
+    itertools.chain.from_iterable(
+        integration_registry.select_integration_requirements(
+            integration_name=integration,
+            target_os=OperatingSystemType.LINUX,
+        )
+        for integration in required_integrations
+    )
+)
+```
+
+#### Any project specific requirements
+
+Any other dependencies that your project relies on. You can then install all of these different requirements through a line in your Dockerfile that looks like the following. It assumes you have accumulated all the requirements in one file.
+
+```Dockerfile
+RUN pip insatll <ANY_ARGS> -r FILE
+```
+
+#### Any system packages
+
+If you have any apt packages that are needed for your application to function, be sure to include them too. This can be achieved in a Dockerfile as follows:
```

Comment:
```suggestion
If you have any `apt` packages that are needed for your application to function, be sure to include them too. This can be achieved in a `Dockerfile` as follows:
```